crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -33,13 +33,8 @@ impl Epoch {
     }
 
     /// Validator related properties, including the active validators
-    async fn validator_set(&self) -> Result<Option<ValidatorSet>> {
-        let system_state: NativeSuiSystemStateSummary = bcs::from_bytes(&self.stored.system_state)
-            .map_err(|e| {
-                Error::Internal(format!(
-                    "Can't convert system_state into SystemState. Error: {e}",
-                ))
-            })?;
+    async fn validator_set(&self, ctx: &Context<'_>) -> Result<Option<ValidatorSet>> {
+        let system_state = self.system_state(ctx).await?;
```

Comment:
Nit: prefer to use ctx.data_provider here instead of using the other methods in the graphQL Object impl block